src/npc.h

```diff
@@ -1358,7 +1358,12 @@ class npc : public Character
         // Player orders a friendly NPC to move to this position
         std::optional<tripoint_abs_ms> goto_to_this_pos;
         int last_seen_player_turn = 0; // Timeout to forgetting
-        const item *wanted_item = nullptr;
+
+        //Safe reference to an item at a specific location which can be checked for subsequent invalidation
+        //in case the object is destroyed or more than one creature goes for it.
+        //const item *wanted_item = nullptr;
```

Comment:
You're overdoing it on the comments here (which is better than underdoing it but still should be addressed). The very fact that you're using an item_location basically tells us what you're saying in the comments. You could say just `// make sure to use item_location in case something happens to the item` if you want an abundance of clarity, but I don't think a comment is necessary at alll

Reply:
I haven't seen it documented where it's explained what conditions invalidate an item_location. One would hope that location movement would do it, but if movement is defined as deletion and re-creation then it's the same as for deletion. There's a actually a lot to be said about item_location, but maybe I haven't looked in the right place.